pkg/cloudprovider/baremetal/provider.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2024 The Operating System Manager contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package baremetal
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	providerconfigtypes "github.com/kubermatic/machine-controller/pkg/providerconfig/types"
+	"k8c.io/operating-system-manager/pkg/cloudprovider/baremetal/types"
+	"k8c.io/operating-system-manager/pkg/providerconfig/config"
+)
+
+func GetCloudConfig(pconfig providerconfigtypes.Config) (string, error) {
+	c, err := getConfig(pconfig)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse config: %w", err)
+	}
+
+	s, err := c.ToString()
+	if err != nil {
+		return "", fmt.Errorf("failed to convert cloud-config to string: %w", err)
+	}
+
+	return s, nil
+}
```

Comment:
Why do we need a cloud-config here if it was empty in machine-controller? https://github.com/kubermatic/machine-controller/blob/main/pkg/cloudprovider/provider/baremetal/provider.go#L216

How would it really come into affect since no-cloud providers usually don't have cloud-configs

Reply:
I delete it until my draft PR in MC is merged